Track5/LinearLesson.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+
+// Lesson 1 - build out linear search
+
+func linearSearch<T: Equatable>(allOf list: [T], forItem item: T) -> Int? {
+  for (index, object) in list.enumerated() where object == item {
```

Comment:
Small nitpick, but `value` might be better than `object` here, because it's likely that we're searching an array of structures rather than classes and `object` would indicate an instance of a class.